service/history/api/signalwithstartworkflow/signal_with_start_workflow.go

```diff
@@ -109,46 +109,48 @@ func startAndSignalWorkflow(
 		return "", err
 	}
 
-	casPredicate, currentWorkflowMutationFn, err := startAndSignalWorkflowActionFn(
+	workflowUpdateFn, err := createWorkflowUpdateFunction(
```

Comment:
```suggestion
	currentWorkflowUpdateFn, err := createWorkflowUpdateFunction(
```

Reply:
@yycptt quick question ... per my understanding of the code, the function returned by the createWorkflowUpdateFunction will not update the current workflow. It will update the workflow provided as an argument to the function. Is this understanding correct?

Reply:
hmm not quite sure if I follow. 

The returned function will be used to update (in startAndSignalWithCurrentWorkflow below) the workflow pointed by `currentWorkflowContext`, so this update is meant for the "currentWorkflow", hence I suggested the naming.